drgn/helpers/linux/slab.py

```diff
@@ -321,3 +333,66 @@ def find_containing_slab_cache(  # type: ignore  # Need positional-only argument
         return slab.slab_cache
     except AttributeError:
         return NULL(prog, "struct kmem_cache *")
+
+
+def get_slab_aliases(prog: Program) -> Dict[str, str]:
+    """
+    Return a dict mapping slab cache name to the cache it was merged with.
+
+    The SLUB subsystem can merge caches with similar settings, as described in
+    the documentation of :func:`slab_cache_is_merged()`. However, if the SLUB
+    sysfs interface is enabled, then the slab merging is exposed as symlinks
+    between directories. So it is possible to use these data structures to
+    discover whether a given slab cache has been merged, and if so, which slab
+    cache it was merged with.
+
+    This function returns a mapping from original cache name, to merged cache
+    name. You can use this mapping to discover the correct cache to lookup via
+    :func:`find_slab_cache()`. If a slab cache name does not appear as a key in
+    this mapping, then it has not been merged with any other cache, and can
+    directly be looked up. A cache name may be mapped to itself; in this case,
+    the cache was merged with others, and the final merge cache used that
+    particular name.
```

Comment:
I'm pondering whether we should include these names that map to themselves. It seems like their presence might cause confusion. You can already tell that a slab cache was "merged into" by its presence in `.values()`.

The only exception is caches that _only_ show up as a mapping to themselves. Do you know how that happens?

Reply:
Yes, even in the vmtest kernels, the returned mapping contains several caches that only show up as a mapping to themselves. It's easier to see in the reverse mapping (final cache -> list of names merged with it). Here's a sample:
```
'kmalloc-8': ['kmalloc-8'],
'kmalloc-8192': ['kmalloc-8192'],
'kmalloc-96': ['blkdev_ioc', 'kmalloc-96'],
'mnt_cache': ['mnt_cache', 'ip6_dst_cache'],
'net_namespace': ['net_namespace'],
'numa_policy': ['numa_policy'],
'pid_namespace': ['pid_namespace'],
'posix_timers_cache': ['posix_timers_cache'],
'shared_policy_node': ['shared_policy_node', 'fasync_cache', 'nsproxy', 'ip_fib_trie'],
'signal_cache': ['RAW', 'PING', 'signal_cache', 'mm_struct'],
```

It's easy enough to see this on a running kernel with sysfs too:
```
ls -lh /sys/kernel/slab | grep -- '->' | awk '{print $NF}' | sort | uniq -c | sort -nr
```
This will show tons of slab IDs which are symlinked to by just one name. I believe all of those slabs are "mergeable", but no matching caches were actually created, so they're left with just the one name. The caches which exist just as a name, with no link, are those which are "unmergeable" due to some setting -- see [sysfs_slab_add()](https://elixir.bootlin.com/linux/v6.0.1/source/mm/slub.c#L5949).

So I think what I'll do with this is call that out in the docstring. I guess the other option would be to filter this mapping so that it only contains caches which were _actually merged_, but I don't think it's worth having the extra logic just to remove some info. Let me know if you disagree.